### PR TITLE
Start review of java events

### PR DESCRIPTION
**Pull request by ChrisChinchilla**
Started a review here @craigwilliams84 as it's not quite finished, I may review the last few sections again later.

I'd love to be able to test the code, do I need to wait for part 1 to be able to do that? Or can I do it now by filling in a few gaps myself?

### CODE REVIEW

**Comment by ChrisChinchilla** [comment]
Oh, most of these are grammar changes and formatting, might look like lots of changes, but just small things really.

**Comment by craigwilliams84** [comment]
> Started a review here @craigwilliams84 as it's not quite finished, I may review the last few sections again later.
> 
> I'd love to be able to test the code, do I need to wait for part 1 to be able to do that? Or can I do it now by filling in a few gaps myself?

Sorry, I've only just noticed this.  You'll probably need a complete picture to test the code because it will involve deploying a smart contract, and sending transactions with web3j.

**Comment by craigwilliams84** [comment]
Comment from @ajsutton over slack RE pantheon_testing.md:

"I’ve run through the example and it works quite well.  The only issue I had is that JUnit 4 wanted `private static void stopPantheon()` to be a public method.

I made a couple of tweaks to mine - instead of setting env variables for configuration, I just used command line arguments and I wound up using dynamically assigned ports so tests can run in parallel without getting conflicts.  So it became:

```
private static GenericContainer pantheonContainer;
  private static Web3j web3j;

  @BeforeClass
  public static void startPantheon() throws InterruptedException {
    pantheonContainer = new GenericContainer("pegasyseng/pantheon:latest");
    pantheonContainer.waitingFor(Wait.forListeningPort());

    pantheonContainer
        .withExposedPorts(8545, 8546)
        .withCommand(
            "--miner-enabled",
            "--miner-coinbase=00a329c0648769a73afac7f9381e08fb43dbea72",
            "--rpc-http-enabled",
            "--rpc-ws-enabled",
            "--network=dev")
        .start();
    final Integer jsonRpcPort = pantheonContainer.getMappedPort(8545);
    web3j = Web3j.build(new HttpService("http://localhost:" + jsonRpcPort));
    waitForPantheonToStart(10000, web3j);
  }

  @AfterClass
  public static void stopPantheon() {
    pantheonContainer.stop();
    web3j.shutdown();
  }
```
"

**Comment by craigwilliams84** [comment]
Comment from @ajsutton over slack RE pantheon_testing.md:

"And, having now read the docs on testcontainers we can go a bit further by using a `ClassRule`:

```
  @ClassRule
  public static final GenericContainer pantheonContainer =
      new GenericContainer("pegasyseng/pantheon:latest")
          .withExposedPorts(8545, 8546)
          .withCommand(
              "--miner-enabled",
              "--miner-coinbase=00a329c0648769a73afac7f9381e08fb43dbea72",
              "--rpc-http-enabled",
              "--rpc-ws-enabled",
              "--network=dev")
          .waitingFor(Wait.forHttp("/liveness").forStatusCode(200).forPort(8545));

  private Web3j web3j;

  @Before
  public void setupWeb3j() {
    final Integer jsonRpcPort = pantheonContainer.getMappedPort(8545);
    web3j = Web3j.build(new HttpService("http://localhost:" + jsonRpcPort));
  }

  @After
  public void stopWeb3j() {
    web3j.shutdown();
  }
```

**Comment by ChrisChinchilla** [comment]
@craigwilliams84 OK, ideally we want these tutorials to build upon each other and allow readers to follow along, so I'll wait for the other part and revisit this one later.

File: java/events.md [hunk 1]
```diff
@@ -2,76 +2,78 @@
 
 ## What is a Smart Contract Event
 
-Events can be emitted by all smart contract functions that are triggered by a transaction, and are an important piece of the Ethereum application architecture puzzle.
+All smart contract functions triggered by transactions emit events, and are an important piece of the Ethereum application architecture puzzle.
```
Comment (craigwilliams84) [inline java/events.md]: Smart contract functions that are triggered by transactions CAN emit events, but not all of them do, so this sentence needs to be modified a bit.

**Review by craigwilliams84 — COMMENTED** [review]
I'm happy with everything else you've suggested!

File: java/events.md (hunk 1 above)
Comment (ChrisChinchilla) [reply java/events.md]: Hows about this change?

Comment (craigwilliams84) [reply java/events.md]: Add a 'they' to "...and `they` are an important piece..." and we've got a deal! :D

Comment (ChrisChinchilla) [reply java/events.md]: So many words!

**Review by craigwilliams84 — APPROVED** [review]

**Pull request merged**